coffee-chats/src/main/java/com/google/step/coffee/tasks/RequestMatcher.java

```diff
@@ -0,0 +1,86 @@
+package com.google.step.coffee.tasks;
+
+import com.google.step.coffee.data.RequestStore;
+import com.google.step.coffee.entity.ChatRequest;
+import com.google.step.coffee.entity.TimeSlot;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet triggered as cron job to match current chat requests together. Triggered periodically by
+ * fetch.
+ */
+@WebServlet("/tasks/request-matching")
+public class RequestMatcher extends HttpServlet {
+
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+      throws ServletException, IOException {
+    RequestStore requestStore = new RequestStore();
+
+    List<ChatRequest> requestList = requestStore.getUnmatchedRequests();
+
+    matchRequests(requestList, requestStore);
+  }
+
+  /**
+   * Take current list of chat requests and store any matched requests into datastore.
+   *
+   * @param requestList list of currently un-matched chat requests from datastore.
+   * @param requestStore interface to datastore to handle requests.
+   */
+  private void matchRequests(List<ChatRequest> requestList, RequestStore requestStore) {
```

Comment:
I don't see tests for this

Reply:
As this is effectively just a placeholder function for what will eventually match requests based on tags, I thought it was redundant to write tests for a function which wasn't functionally correct yet.